app/ui-flextab/client/tabs/uploadedFilesList.js

```diff
@@ -152,6 +157,71 @@ Template.uploadedFilesList.events({
 			},
 		};
 
+		const canDelete = () => {
+			console.log(2);
+			const forceDelete = hasAtLeastOnePermission('force-delete-message', this.rid);
+			const hasPermission = hasAtLeastOnePermission('delete-message', this.rid);
+			const isDeleteAllowed = settings.get('Message_AllowDeleting');
+			const deleteOwn = this.file.userId === Meteor.userId();
+			if (!(hasPermission || (isDeleteAllowed && deleteOwn) || forceDelete)) {
+				return;
+			}
+			console.log(3);
+			const blockDeleteInMinutes = settings.get('Message_AllowDeleting_BlockDeleteInMinutes');
+			if (forceDelete) {
+				return true;
+			}
+			console.log(4);
+			if (blockDeleteInMinutes != null && blockDeleteInMinutes !== 0) {
+				let msgTs;
+				if (this.file.uploadedAt != null) {
+					msgTs = moment(this.file.uploadedAt);
+				}
+				let currentTsDiff;
+				if (msgTs != null) {
+					currentTsDiff = moment().diff(msgTs, 'minutes');
+				}
+				console.log(5);
+				return currentTsDiff < blockDeleteInMinutes;
+			} else {
+				console.log(6);
+				return true;
+			}
+		};
+
+		if (canDelete()) {
+			config.columns[0].groups[0].items.push({
```

Comment:
There is a better way to inject this item **inside the declaration** of `config`:

```js
const config = {
	columns: [
		{
			groups: [
				{
					items: [
						{
							icon: 'download',
							// something something
						},
						// Using spread operator here does the trick
						...(canDelete ? [{
							icon: 'trash',
							// something something
						}] : []),
					],
				},
			],
		},
	],
	currentTarget: e.currentTarget,
	onDestroyed:() => {
		e.currentTarget.parentElement.classList.remove('active');
	},
};
```

Reply:
Now this is too new for me 😅 thanks!!